### ISSUE TITLE
Downloaded file corrupted?

### ISSUE THREAD

**Issue by ianpgm** [invenia/FTPClient.jl#20]
I'm trying to use FTPClient.jl to download a file, but the file seems to be corrupted (i.e. gunzip refuses to open it complaining the format is wrong), unless I'm using FTPClient incorrectly (which is more likely). This is what I'm trying to do in the REPL (Julia version 0.4.1, MacOS X 10.11.3) :

```
using FTPClient

ftp_init()
0
ftp_options = RequestOptions(url="ftp.ebi.ac.uk/pub/databases/Pfam/current_release/")
FTPClient.RequestOptions(true,false,false,true,false,"","","ftp.ebi.ac.uk/pub/databases/Pfam/current_release/","localhost",true)

ftp_get("Pfam-A.hmm.gz", ftp_options, "Pfam-A.hmm.gz")
Response Code :226
Request Time  :22.733693
Headers       :AbstractString["220-   ","220- ftp1.ebi.ac.uk FTP server","220-   ","220- WARNING: please note that the private part of this ftp service ","220- has been migrated to ftp-private.ebi.ac.uk on 3rd June 2010.","220 ","331 Please specify the password.","230 Login successful.","257 \"/\"","250 Directory successfully changed.","250 Directory successfully changed.","250 Directory successfully changed.","250 Directory successfully changed.","229 Entering Extended Passive Mode (|||11572|).","200 Switching to ASCII mode.","150 Opening BINARY mode data connection for Pfam-A.hmm.gz (254622237 bytes).","226 Transfer complete."]
Length of body: 254609678

ftp_cleanup()
exit()
```

Then from the command line:

```
$ gunzip Pfam-A.hmm.gz
gzip: Pfam-A.hmm.gz: invalid compressed data--format violated

```

Downloading the same file with curl works fine:

```
$ curl ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz -o Pfam-A.hmm.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  242M  100  242M    0     0  12.2M      0  0:00:19  0:00:19 --:--:-- 14.2M

gunzip Pfam-A.hmm.gz

```

I would be grateful if you could let me know whether I'm doing something wrong or whether there's some bug here. Thanks!

**Comment by samuel-massinon** [comment]
It looks like you are doing it right. I tried it myself and I got the same problem. I'll look into it more tomorrow.

**Comment by samuel-massinon** [comment]
Well that was a doozy of a bug. Took me nearly two days but I finally nailed it down. Looks like we were downloading the file in ASCII mode instead of binary mode, which causes some issues. We do have some binary options, but `ftp_get` doesn't seem to listen to them. I was able to find a way to make `ftp_get` use binary mode but it will require some code change, which I can hopefully get to you asap.

**Comment by samuel-massinon** [comment]
@ianpgm if you need this FTPClient asap, you can use this branch https://github.com/invenia/FTPClient.jl/tree/binary_downloading

Once I merge it into master and push a new version to METADATA I'll update you again and close this issue. 

Thanks for finding this bug!

**Comment by ianpgm** [comment]
Thanks! It's not so urgent that I can't wait for the master version.

**Comment by kano31** [comment]
resolved with #21
